src/Entity/EntityIdValue.php

```diff
@@ -58,10 +58,14 @@ private function getNumericId() {
 	 * @throws IllegalValueException
 	 */
 	public function unserialize( $serialized ) {
-		list( $entityType, $numericId ) = json_decode( $serialized );
+		list( $type, $id ) = json_decode( $serialized );
 
 		try {
-			$entityId = LegacyIdInterpreter::newIdFromTypeAndNumber( $entityType, $numericId );
+			if ( is_string( $id ) ) {
```

Comment:
If the class doesn't exist, this should fail hard. Ok, we could throw an exception instead of a fatal error.